tripy/tripy/backend/compiler_api.py

```diff
@@ -80,20 +80,25 @@ def __init__(
                         "Shape values must be numbers.",
                         [f"Shape: {shape} contains an element: {repr(elem)} with non-numerical value(s)"],
                     )
-                if len(elem) != 3:
-                    raise_error(
-                        "Incorrect number of shape values provided.",
-                        [
-                            f"Exactly 3 shape values must be provided for each dimension (min/opt/max)"
-                            f" but got: {len(elem)} values in shape: {shape}. "
-                        ],
-                    )
+            elif isinstance(elem, Shape):
+                elem = elem.data().data()
```

Comment:
This is not suboptimal I guess. We should be able to slice shape tensors and populate shape bounds.

Reply:
Why is `elem` a shape tensor? What you want is the shape itself be a shape tensor describing the shape of input. 
```
a = tp.ones((2,3))
shape_of_a = a.shape
a_info = tp.InputInfo(shape_of_a, dtype=tp.float32) 
```
Your current approach is doing:
```
shape_of_a = a.shape
a_info = tp.InputInfo([shape_of_a[0], shape_of_a[1]], dtype=tp.float32) 
```